### PR TITLE
general: add astyle check to travis

### PR DESCRIPTION
**Pull request by amiclaus**
Checks modified/new source/header files from pull request to determine if they are
properly formatted.

Also, create separate script file for Travis

Signed-off-by: Antoniu Miclaus <antoniu.miclaus@analog.com>

### CODE REVIEW

File: .travis_do [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+fi
+
+FILES=$(git diff --name-only $COMMIT_RANGE)
+
+is_source_file() {
+	local file="$1"
+	local ext_list=".c .h"
+
+	for ext in $ext_list; do
+		[[ ${file: -2} == $ext ]] && return 0
```
Comment (commodo) [inline .travis_do]: it's a good idea to use quotes in shell where possible; otherwise this statement can fail if ` ${file: -2}` expands to empty

Comment (commodo) [reply .travis_do]: i.e. `[ "${file: -2}" == "$ext" ]`   ; 
single brackets can be used here too

File: .travis_do [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+for file in $FILES; do
+	if is_source_file $file
+	then 
+		astyle --options=./astyle_config $file
+	fi
+done;
+
+git diff --name-only --exit-code || {
```
Comment (commodo) [inline .travis_do]: i'd remove the `--name-only` part ; showing the diff is useful here

File: CI/travis/before_install_darwin [hunk 3]
```diff
@@ -10,4 +10,5 @@ brew_install_or_upgrade() {
 }
 
 brew_install_or_upgrade libmatio
-brew_install_or_upgrade cppcheck
\ No newline at end of file
+brew_install_or_upgrade cppcheck
+brew_install_or_upgrade astyle
```
Comment (commodo) [inline CI/travis/before_install_darwin]: at this point, it's a good idea to switch to a for loop
have a list of packages and iterate
```
PACKAGES="libmatio cppcheck astyle"
for package in $PACKAGES ; do
    brew_install_or_upgrade "$package"
done
```

**Comment by amiclaus** [comment]
Changelog v1->v2:
 * minor improvements based on code review

File: .travis_do [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+for file in $FILES; do
+	if is_source_file $file
+	then 
+		astyle --options=./astyle_config $file
+	fi
+done;
+
+git diff --exit-code || {
+	echo_red "AStyle needs to ne applied"
```
Comment (commodo) [inline .travis_do]: this should be something more like  `echo_red "Code has style inconsistencies...`

**Comment by amiclaus** [comment]
Changelog v2->v3:
 * Change code style error message

File: .travis_do [hunk 5]
```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+set -e
+
+echo_red() { printf "\033[1;31m$*\033[m\n"; }
+
+if [ "$TRAVIS_OS_NAME" == "linux" ]
```
Comment (commodo) [inline .travis_do]: thinking about this a bit, maybe it's a good idea [if moving everything to script files] to have them functionally separate; 

i.e. how a `astyle.sh` script that does only astyle logic, another script that handles builds (let's say `build_projects.sh`)

a monolithic script becomes hard to follow as it grows over time ;

for `cppcheck` you could move it into a script as well, but maybe later ; it's only one line in travis, so not sure if it's worth having it into it's own script file

**Comment by amiclaus** [comment]
Changelog v3 -> v4:
 * Split into multiple scripts based on functionality

**Comment by commodo** [comment]
Will merge later today if no objections.

**Comment by dbogdan** [comment]
I would move the scripts to a dedicated folder (/CI/scripts or maybe /scripts since they can be run manually too ?).
I would also rename CI to ci, but this is only a personal preference and I don't mind if it will rename the same :). Windows and Mac OS are not differentiating the files or folders names by case (unlike Linux) so I'm trying to not use uppercase very often.

**Comment by commodo** [comment]
Will defer merging until comments have been resolved

**Comment by amiclaus** [comment]
Changelog v4 -> v5:
 * move scripts to folder
 * use lowercase letters

File: .travis.yml [hunk 6]
```diff
@@ -11,10 +11,10 @@ compiler:
...
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then ./ci/travis/before_install_darwin ; fi
+  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then ./ci/travis/before_install_linux ; fi
 
 script:
-  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then make -C ${TRAVIS_BUILD_DIR}/ad9361/sw/ -f Makefile.linux ; fi
-  - make -C ${TRAVIS_BUILD_DIR}/ad9361/sw/ -f Makefile.generic
-  - cppcheck --quiet --force --error-exitcode=1 .
+  - ./ci/scripts/build_projects.sh
+  - ./ci/scripts/cppcheck.sh
+  - ./ci/scripts/astyle.sh
```
Comment (amiclaus) [inline .travis.yml]: When calling the scripts, the path (home dir) does not change. Actually, I think `cppcheck --quiet --force --error-exitcode=1 ../../.` should be reverted to point to the current dir.

Comment (dbogdan) [reply .travis.yml]: You are right, I missed that - sorry.

**Comment by amiclaus** [comment]
Changelog v5 -> v6: 
 * change cppcheck path to current dir.

**Comment by commodo** [comment]
Will merge this later today if no objections.
